### ISSUE TITLE
`avoid-leaking-state-in-ember-objects` false positives in mixins

### ISSUE THREAD

**Issue by boris-petrov** [ember-cli/eslint-plugin-ember#1116]
Originally posted [here](https://github.com/ember-cli/eslint-plugin-ember/pull/1081#issuecomment-804702651).

I believe this is missing two cases:

```typescript
Mixin.create({
  // both get the warning and both shouldn't
  foo: undefined as string | undefined,
  bar: undefined as any,
})
```

And service-injections:

```typescript
Mixin.create({
  store: (inject() as unknown) as Store,
})
```

All of these get a `avoid-leaking-state-in-ember-objects` warning but all of them are fine I believe.

cc @bmish, @jaydgruber
